cmd/go-httpbin/main.go

```diff
@@ -64,7 +82,25 @@ func main() {
 		httpbin.WithObserver(httpbin.StdLogObserver(logger)),
 	)
 
-	listenAddr := net.JoinHostPort("0.0.0.0", strconv.Itoa(port))
-	logger.Printf("go-httpbin listening on %s", listenAddr)
-	http.ListenAndServe(listenAddr, h.Handler())
+	listenAddr := net.JoinHostPort(host, strconv.Itoa(port))
+
+	server := &http.Server{
+		Addr:    listenAddr,
+		Handler: h.Handler(),
+	}
+
+	if httpsCertFile != "" && httpsKeyFile != "" {
+		cert, err := tls.LoadX509KeyPair(httpsCertFile, httpsKeyFile)
+		if err != nil {
+			log.Fatal("Failed to generate https key pair: ", err)
+		}
+		server.TLSConfig = &tls.Config{
+			Certificates: []tls.Certificate{cert},
+		}
+		logger.Printf("go-httpbin listening on %s with https", listenAddr)
```

Comment:
Minor, but for usability's sake maybe

```go
logger.Printf("go-httpbin listening on https://%s", listenAddr)
```

would be better?  If nothing else, it would allow folks whose terminals will automagically recognize URLs to click on the link directly?

(Same suggestion would apply to the plain HTTP case below, too!)

Reply:
Sure - will do that and update the thread.

Reply:
@mccutchen updated the message and squashed.